### ISSUE TITLE
UnnecessaryFullyQualifiedName false positive with name collision

### ISSUE THREAD

**Issue by Gama11** [detekt/detekt#9017]
The `UnnecessaryFullyQualifiedName` rule newly added in 2.0.0-alpha.2 reports a false positive in this scenario:

```kotlin
sealed interface Foo<T> {
    data object Any : Foo<kotlin.Any>
}
```

If the `kotlin.` prefix is removed, the type parameter refers to `Foo.Any`, meaning the qualifier is not unnecessary.

**Comment by Gama11** [comment]
On second thought, this one may not be considered a bug, if the intent is simply to force the use of an `import as`.

**Comment by atulgpt** [comment]
> On second thought, this one may not be considered a bug, if the intent is simply to force the use of an import as.

If this is the case then it should be under flag with default false(i.e., not reporting name collision cases)

**Comment by BraisGabin** [comment]
> On second thought, this one may not be considered a bug, if the intent is simply to force the use of an `import as`.

It is a false positive. And I don't think that we should add any flag, just fix it. The issue is open to grab by anyone willing to fix it.

**Comment by mirkoalicastro** [comment]
Hi @BraisGabin. I've raised a PR to fix this (and couple of others) false positives: https://github.com/detekt/detekt/pull/9223.
Could you have a look?
